lib/spack/spack/package.py

```diff
@@ -639,26 +639,53 @@ def do_fetch(self):
                     "Will not fetch %s." % self.spec.format('$_$@'), checksum_msg)
 
         self.stage.fetch()
+
+        ##########
+        # Fetch resources
+        resources = self._get_resources()
+        # FIXME : choose the unique name appropriately. Is there a function somewhere for the base name ?
+        pieces = [self.name, str(self.version), self.spec.dag_hash()]
+        for resource in resources:
+            resource_stage_folder = '-'.join(pieces + [resource.name])
```

Comment:
I think you want something like `'resource-' + <resource_name> + self.spec.dag_hash()` -- that would a) separate resource stages from other ones, b) differentiate by resource names within a particular configuration's build, and c) differentiate between resources fetched for all the different DAG versions.

`self.stage.path` should get you the stage's path, which is the parent of `self.stage.source_dir`